File: 0066.加一/0066-加一.java
class Solution {
    public int[] plusOne(int[] digits) {
        // 加1，求10的模，不为0则没有发生进位，直接返回
        for(int i = digits.length - 1; i >= 0; i--){
            digits[i] = digits[i] + 1;
            digits[i] = digits[i] % 10;
            if(digits[i] != 0){
                return digits;
            }
        }
        // 当都是9时，长度加1，最高位为1，其余为0
        digits = new int[digits.length + 1]; // 默认为0，可以不用再次赋值
        digits[0] = 1;
        return digits;
    }
}

File: 0066.加一/0066-加一.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* plusOne(int* digits, int digitsSize, int* returnSize){
    int flag = 0;
    int i = digitsSize - 1;
    while((++digits[i]) == 10)
    {
        digits[i] = 0;
        if(i == 0)
        {
            flag = 1;
            break;
        }
        //digits[i - 1]++;
        i--;
    }

    if(flag == 1)
    {
        digitsSize++;
        int *newDigits = malloc(sizeof(int) * digitsSize);
        newDigits[0] = 1;
        for(int i = 1; i < digitsSize; i++)
        {
            newDigits[i] = digits[i - 1];
        }
        digits = newDigits;
    }

    *returnSize = digitsSize;
    return digits;
}